kong/plugins/oidc/handler.lua

```diff
@@ -29,25 +30,44 @@ end
 
 function handle(oidcConfig, oidcSessionConfig)
   local response
+
+  -- attempt introspection of potential bearer token
   if oidcConfig.introspection_endpoint then
     response = introspect(oidcConfig)
+    -- if response, then introspect successful
     if response then
-      utils.injectUser(response)
+      local access_token = utils.get_bearer_access_token_from_header(oidcConfig)
+      local user_info, err = get_userinfo(oidcConfig, response)
+      
+      -- introspect passed but userinfo call failed, maybe we have id_token
+      if err then
+        ngx.log(ngx.DEBUG, "call to userinfo endpoint failed, attaching decoded token to user")
+        user_info = response
+      end
+      
+      response = {
+        access_token = access_token,
```

Comment:
add comment about convention started/assumption - we inject token used as access token even though it could be id_token

Reply:
add todo how do we distinguish between access token and id token?